src/Alley/Validator/DivisibleBy.php

```diff
@@ -32,41 +32,22 @@ final class DivisibleBy extends ExtendedAbstractValidator
         'divisor' => 1,
     ];
 
-    private ValidatorInterface $validDivisors;
-
-    private ValidatorInterface $validRemainders;
-
-    public function __construct($options)
-    {
-        $this->validDivisors = new Comparison([
-            'compared' => 0,
-            'operator' => '!==',
-        ]);
-        $this->validRemainders = new Comparison(
-            [
-                'compared' => 0,
-                'operator' => '===',
-            ],
-        );
-
-        parent::__construct($options);
-    }
-
     protected function testValue($value): void
     {
-        if (! $this->validRemainders->isValid((int) $value % $this->options['divisor'])) {
+        $value = (int) $value;
+        $actual = $value % $this->options['divisor'];
+
+        if ($actual !== 0) {
             $this->error(self::NOT_DIVISIBLE_BY);
         }
     }
 
     protected function setDivisor($divisor)
     {
         $divisor = (int) $divisor;
-        $valid = $this->validDivisors->isValid($divisor);
 
-        if (! $valid) {
-            $messages = $this->validDivisors->getMessages();
-            throw new InvalidArgumentException("Invalid 'divisor': " . current($messages));
+        if ($divisor === 0) {
```

Comment:
do we need to check if the divisor is an int as well? Or is that handled somewhere else?

Reply:
The divisor can also be a float or a string or whatever else PHP can convert to an integer. I'm kinda inclined to not get in the way of that. Also, I don't want to introduce a restriction with this change that would require a major version bump.